tests/basic.js

```diff
@@ -72,11 +72,40 @@ tape('fetch', function (t) {
             t.ok(d.subtitle, 'we have a subtitle');
             t.ok(d.trailer, 'we have a trailer');
             t.ok(d.synopsis, 'we have a synopsis');
-            t.ok(d.type===Generic.TabType.MOVIE || d.type===Generic.TabType.TVSHOW || d.type===Generic.TabType.ANIME, 'we have a type field w');
+            t.ok(d.type===Provider.TabType.MOVIE || d.type===Provider.TabType.TVSHOW || d.type===Provider.TabType.ANIME, 'we have a type field which is a tab type');
         });
 
         t.end();
     }).catch(function (e) {
         t.notOk(e, 'failed fetch');
     });
 });
+
+tape('random', function (t) {
+    debug('random, timeout', config.timeout);
+    t.timeoutAfter(config.timeout);
+
+    var P = load();
+    var I = new P(config.args);
+
+    I.random().then(function (r) {
+        debug ('random', r);
+        t.ok(r, 'we were able to get a random');
+        t.ok(r[I.config.uniqueId] || d.id, 'we have an unique id');
+        t.ok(r.title, 'we have a title');
+        t.ok(r.year, 'we have a year');
+        t.ok(r.genre, 'we have a genre field');
+        t.ok(r.genre.length > 0, 'we have at least 1 genre');
+        t.ok(r.rating, 'we have a rating');
+        t.ok(r.poster, 'we have a poster');
+        t.ok(r.backdrop, 'we have a backdrop');
+        t.ok(r.torrents, 'we have a torrents field');
+        t.ok(r.subtitle, 'we have a subtitle');
+        t.ok(r.trailer, 'we have a trailer');
+        t.ok(r.synopsis, 'we have a synopsis');
+        t.ok(r.type===Provider.TabType.MOVIE || r.type===Provider.TabType.TVSHOW || r.type===Provider.TabType.ANIME, 'we have a type field which is a tab type');
```

Comment:
as we're basically testing the same as detail, why not factor that out in the same function ?